dags/multipod/configs/maxtext_sweep_gce_config.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to construct job sweep GCE configs for maxtext DAG."""
+
+from xlml.apis import gcp_config, metric_config, task, test_config
+from dags.vm_resource import TpuVersion
+import datetime
+import itertools
+from typing import List, Iterable
+
+
+def get_maxtext_sweep_gce_config(
+    test_owner: str,
+    tpu_version: TpuVersion,
+    num_slices: List[int],
+    sweep_params: {},
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    run_name_prefix: str,
+    project_name: str,
+    runtime_version: str,
+    base_set_up_cmds: Iterable[str],
+    base_run_model_cmds: Iterable[str],
+    is_tpu_reserved: bool = True,
+    network: str = "default",
+    subnetwork: str = "default",
+) -> List[task.TpuQueuedResourceTask]:
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.XLML_DATASET,
+      dataset_project=project_name,
+      composer_project=project_name,
+  )
+
+  # Add num_slices as a sweep param
+  sweep_params["NUM_SLICES"] = num_slices
+
+  # Convert sweep_params to a list of lists to generate sweep param combinations
+  sweep_params_list = []
+  for param, values in sweep_params.items():
+    sweep_params_list.append([(param, val) for val in values])
+
+  # Generate all combinations of sweep param configurations and create a TpuQueuedResourceTask for each one
+  qr_task_list = []
+  current_datetime = datetime.datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+  for idx, config in enumerate(itertools.product(*sweep_params_list)):
+    config_dict = {key: value for (key, value) in config}
+
+    # Remove num_slices as a sweep param after combinations have been generated
+    curr_num_slices = config_dict["NUM_SLICES"]
+    del config_dict["NUM_SLICES"]
+
+    # Add MaxText run_name
+    run_name = f"{run_name_prefix}-{curr_num_slices}x{tpu_version.value}-{tpu_cores}-{current_datetime}-{idx}"
```

Comment:
Do we need `{idx}` here? It seems like that `{current_datetime}` already makes the `run_name` unique.

Reply:
`idx` is needed here since multiple MaxText jobs can be launched and `current_datetime` will be the same for all of them.

Reply:
Oh i see! It's probably good to not use parameters either since there could be a lot of parameters to sweep.